core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -793,17 +799,33 @@ public boolean isStoreWidth() {
         return storeWidth;
     }
 
+
+    public DataFlagEncoder setSpatialRules(int rules) {
+        this.spatialRules = rules;
+        return this;
+    }
+
+    public DataFlagEncoder setSpatialRuleLookup(SpatialRuleLookup spatialRuleLookup) {
+        if (spatialRuleLookup.size() != spatialRules)
+            throw new IllegalArgumentException("You have to configure the encoder to accept the identical amount of rules " +
```

Comment:
Should we get the number of spatial rules from the lookup then? But the lookup isn't available when constructing the encoder?